pkg/v1/sdk/capabilities/discovery/cluster_test.go

```diff
@@ -141,3 +148,239 @@ func TestClusterQueries(t *testing.T) {
+		},
+		{
+			description: "group found, resource not found",
+			query:       Group("test", "autoscaling").WithResource("verticalautoscaling"),
+			want:        false,
+		},
+		{
+			description: "group and resource found",
+			query:       Group("test", "autoscaling").WithResource("horizontalpodautoscalers"),
+			want:        true,
+		},
+		{
+			description: "group exists, but empty string resource returns error",
+			query:       Group("test", "autoscaling").WithResource(""),
+			err:         ErrInvalidWithResourceMethodArgument,
+		},
+		{
+			description: "group not found, version found",
+			query:       Group("test", "hotscaling").WithVersions("v1"),
+			want:        false,
+		},
+		{
+			description: "group and version found",
+			query:       Group("test", "autoscaling").WithVersions("v1"),
+			want:        true,
+		},
+		{
+			description: "group and versions found",
+			query:       Group("test", "autoscaling").WithVersions("v2beta1", "v1", "v2beta2"),
+			want:        true,
+		},
+		{
+			description: "group found, but empty string version returns error",
+			query:       Group("test", "autoscaling").WithVersions(""),
+			err:         ErrInvalidWithVersionsMethodArgument,
+		},
+		{
+			description: "core group and version found",
+			query:       Group("test", "").WithVersions("v1"),
+			want:        true,
+		},
+		{
+			description: "group found and version not found",
+			query:       Group("test", "autoscaling").WithVersions("v1", "v2", "v2beta2"),
+			want:        false,
+		},
+		{
+			description: "group, versions, resource found",
+			query: Group("test", "autoscaling").
+				WithVersions("v1", "v2beta1", "v2beta2").
+				WithResource("horizontalpodautoscalers"),
+			want: true,
+		},
+		{
+			description: "group, versions found, resource not found",
+			query: Group("test", "autoscaling").
+				WithVersions("v1", "v2beta1", "v2beta2").
+				WithResource("verticalscaler"),
+			want: false,
+		},
+		{
+			description: "group, resource found, version not found",
+			query: Group("test", "autoscaling").
+				WithVersions("v2").
+				WithResource("horizontalpodautoscalers"),
+			want: false,
+		},
+		{
+			description: "versions, resource found, group not found",
+			query: Group("test", "hotscaling").
+				WithVersions("v2beta1").
+				WithResource("horizontalpodautoscalers"),
+			want: false,
+		},
+		{
+			description: "group, versions, resource not found",
+			query: Group("test", "hotscaling").
+				WithVersions("v3").
+				WithResource("verticalscaler"),
+			want: false,
+		},
+		{
+			description: "multiple empty versions returns error even if resource exists",
+			query: Group("test", "").
+				WithVersions("", "", "", "").
+				WithResource("pods"),
+			err: ErrInvalidWithVersionsMethodArgument,
+		},
+		{
+			description: "having empty string version among many returns an error",
+			query:       Group("test", "").WithVersions("", "v1", "v2", "v3"),
+			err:         ErrInvalidWithVersionsMethodArgument,
+		},
+	}
```

Comment:
I would also like to see test cases for below scenarios as well 
`Group("test", "").WithVersions("", "", "", "").WithResource("")`
`Group("test", "autoscaling").WithVersions("", "", "", "").WithResource("")`

Reply:
Tests added.